frontend/src/pages/RoadDetails.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+import '../css/roadDetails.css'; // Import the CSS file
+
+import MiddleArea from '../Components/roadDetails/MiddleArea';
+import MiddleArea2 from '../Components/roadDetails/MiddleArea2';
+import TopBar from '../Components/roadDetails/TopBar';
+import PlotArea from '../Components/roadDetails/PlotArea';
+
+const RoadDetails = () => {
+  const [showMiddleArea2, setShowMiddleArea2] = useState(false);
```

Comment:
Use a more specific variable name:
```suggestion
  const [isMapShown, setIsMapShown = useState(false);
```